### DISCUSSION TITLE
Raster vs vector basemaps

### DISCUSSION THREAD

**Discussion by HatYaz** [OpenDrift/opendrift#1314] [Q&A]
As I consider transitioning from raster (BSB) charts to vector (S-57) charts, I would like to explore how this change might enhance my opendrift simulations. I have identified several potential benefits and would love to hear your thoughts and experiences:

Enhanced Precision: Have you noticed improvements in the precision of drift predictions, particularly in complex coastal areas, due to the more detailed shoreline and bathymetry data provided by vector charts?

Better Integration: How has the seamless integration of oceanographic and meteorological data with vector charts improved the overall accuracy of your simulations?

Faster Processing: Have you experienced quicker simulation runs and the ability to handle more complex scenarios thanks to the improved computational efficiency of vector charts?

Dynamic Updates: In what ways has the easier incorporation of real-time data updates enhanced the model’s responsiveness to changing conditions in your applications?

I believe these changes could lead to more accurate, efficient, and versatile simulations within OpenDrift, and I am eager to hear your insights and feedback on this potential upgrade.

Thank you for your input!

**Comment by knutfrode** [comment]
ChatGPT is wrong on several things.
Raster is faster.

**Reply by HatYaz** [reply]
Thanks Mark for your short answer.

Joke aside I think the above points are seriously suffisant to worth being answered
For ChatGPT or human style the objective is the same: serving the user with serious (for the 2nd time) and clear answers.

Cheers

**Reply by knutfrode** [reply]
The default in OpenDrift is to use a 500m rasterised version of the global GSHHG shoreline for a pre-screening of elements that are not close to land. The (normally few) elements that are very close to land are then checked if they are inside any of the polygons of the vector version of GSHHG at full resolution. The pre-screening with raster makes it much faster, whereas the accuracy of the full resolution vector coastline is still used in effect.
This should be sufficient for most oceanic applications, even inside fjords.

However, if you have a very narrow fjord or a bay or a lake, you may well use a vector coastline if you have one available. However, the raster coastline normally included with the ocean model that covers this narrow area should then also probably be sufficient (if good enough for the ocean model simulation, it is probably also good enough for the subsequent drift simulation).

There is a dedicated `reader_shape` that you can use with shapefiles:
https://opendrift.github.io/gallery/example_shape_landmask.html

Here you can see the difference in using GSHHG vector (with raster pre-screening) and model landmask, however, in this case the ocean model landmask is very coarse (4 km).
https://opendrift.github.io/gallery/example_model_landmask.html

**Comment by HatYaz** [comment]
Thanks for the answer, 
As previously stated, I can utilize both raster and vector data. I wanted to know about the benefits of employing each type. You've now affirmed that when dealing with a narrow fjord, bay, or lake, a vector coastline could be an appropriate choice.
Cheers
